roles/sap_netweaver_preconfigure/tasks/sapnote/3119751.yml

```diff
@@ -5,23 +5,51 @@
     msg: "SAP note {{ (__sap_netweaver_preconfigure_sapnotes_versions | selectattr('number', 'match', '^3119751$') | first).number }}
           (version {{ (__sap_netweaver_preconfigure_sapnotes_versions | selectattr('number', 'match', '^3119751$') | first).version }}): Linux Requirements for SAP Kernel 754 and for SAP Kernel 788 and higher"
 
-- name: Get info about the compat-sap-c++-10.so file
-  ansible.builtin.stat:
-    path: /opt/rh/SAP/lib64/compat-sap-c++-10.so
-  register: __sap_netweaver_preconfigure_register_stat_compat_sap_cpp
-
-- name: Create directory '{{ sap_netweaver_preconfigure_rpath }}'
-  ansible.builtin.file:
-    path: "{{ sap_netweaver_preconfigure_rpath }}"
-    state: directory
-    owner: root
-    group: root
-    mode: '0755'
-  when: __sap_netweaver_preconfigure_register_stat_compat_sap_cpp.stat.exists
-
-- name: Create a link to '{{ sap_netweaver_preconfigure_rpath }}/libstdc++.so.6'
-  ansible.builtin.file:
-    src: /opt/rh/SAP/lib64/compat-sap-c++-10.so
-    dest: "{{ sap_netweaver_preconfigure_rpath }}/libstdc++.so.6"
-    state: link
-  when: __sap_netweaver_preconfigure_register_stat_compat_sap_cpp.stat.exists
+# Note: This file is only included for RHEL 8, so no further when condition is required here.
+
+# Reason for noqa: We do not want to fail if there is no output from the ls command.
+- name: Identify the latest compat-sap-c++-NUM.so file with NUM >= 10 # noqa risky-shell-pipe
+  ansible.builtin.shell:
+  args:
+    cmd: ls compat-sap-c++-1* | sort | tail -1
+    chdir: /opt/rh/SAP/lib64
+  register: __sap_netweaver_preconfigure_register_ls_compat_sap_cpp
+  changed_when: false
+
```

Comment:
Is there a reason why `ansible.builtin.find` would not work? It also supports regex and powerful filter options, similar to the find command itself.

Reply:
No known reason, fixed now! Thanks for reporting this.